src/DataUtils.cpp

```diff
@@ -0,0 +1,211 @@
+#include "NAS2D/DataUtils.h"
+
+#include "NAS2D/Xml/XmlAttribute.h"
+#include "NAS2D/Xml/XmlElement.h"
+
+#include <algorithm>
+#include <sstream>
+
+#if defined(_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#endif
+
+void NAS2D::Xml::validateXMLElement(
+	const NAS2D::Xml::XmlElement& element,
+	const std::string& name,
+	const std::string& requiredChildElements,
+	const std::string& requiredAttributes,
+	const std::string& optionalChildElements,
+	const std::string& optionalAttributes)
+{
+	if (name.empty())
+	{
+		// TODO: Make this a function with [[noreturn]] attribute
+#if defined(_WIN32)
+		std::ostringstream err_ss;
+		err_ss << "Xml Validation failed. Element name is required.";
+		::MessageBoxA(nullptr, err_ss.str().c_str(), "Fatal Error", MB_ICONERROR | MB_OK);
+#else
+		// TODO: Add non-windows version.
+#endif
+		exit(0);
```

Comment:
Ehh, such code is of very questionable use in a library. People generally don't want their calls to some library to suddenly terminate their app.

The error handling on this really needs to be reworked. I would suggest throwing an exception (`std::runtime_error` or derived) with the desired error message.

Reply:
These are workarounds for not having Leeor's `doNonFatalErrorMessage` and `doAlertMessage` functions available in the engine. I'd really like to add those instead of the above.